Sources/StreamVideo/WebRTC/WebRTCClient.swift

```diff
@@ -1163,10 +1201,93 @@ class WebRTCClient: NSObject, @unchecked Sendable {
         )
     }
     
+    private func subscribeToInternetConnectionUpdates() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleConnectionStateChange),
+            name: .internetConnectionStatusDidChange,
+            object: nil
+        )
+    }
+    
+    @objc private func handleConnectionStateChange(_ notification: NSNotification) {
+        guard let status = notification.userInfo?[Notification.internetConnectionStatusUserInfoKey] as? InternetConnection.Status else {
+            return
+        }
+        
+        if status == .unavailable {
+            if !isFastReconnecting {
+                isFastReconnecting = true
```

Comment:
Here we should probably capture the current timestamp. Once we are back online, we can check the timestamp we captured and compare it with the current timestamp. If the diff is less or equal to the `fastReconnectTimeout` and the connections haven't been disconnected, then we should try the fast reconnect.

TBH the current approach will still work as we rely on the SFU to tell us if we can reconnect, but my comments are align the implementation with Android (and potentially React)

Reply:
Yes, that's a good point, already updated the PR.